src/kernel/capinv.c

```diff
@@ -483,7 +483,8 @@ cap_thd_switch(struct pt_regs *regs, struct thread *curr, struct thread  *next,
 		assert(!(next->state & THD_STATE_PREEMPTED));
 		next->state &= ~THD_STATE_RCVING;
 		thd_state_evt_deliver(next, &a, &b);
-		__userregs_setretvals(&next->regs, thd_rcvcap_pending_dec(next), a, b);
+		thd_rcvcap_pending_dec(next);
+		__userregs_setretvals(&next->regs, thd_rcvcap_pending(next), a, b);
```

Comment:
This is fine for now.  I wonder if it is a problem with the API, or with the usage of the API.  Your fix is for the usage of the API.  Food for thought.